src/FileIO.cpp

```diff
@@ -580,5 +580,24 @@ namespace FileIO {
 
       return (0 == rc);
    }
+
+   Result<std::vector<std::string>> GetDirectoryContents(const std::string& directory) {
```

Comment:
It only looks for entries that are type DT_REG - so no directories should be pushed back.